### PR TITLE
fix(core): fix crash when history does not include valid user information

### PR DESCRIPTION
**Pull request by binoy14**
### Description

When a dataset is copied without history it stores userId as `<system>` which is not a valid userId. This change fixes studio to show a dummy user information when invalid ids are present.

### What to review

- Does the logic make sense? Can it be simplified? 
    -  I created a new variable for storing invalid userIds, it might be a little over engineered for now but it allows flexibility to handle similar cases in the future.

### Notes for release

Fix crash in studio when viewing history for dataset that is copied without history

### CODE REVIEW

**Review by rexxars — CHANGES_REQUESTED** [review]
Nice. I'd want to add a profile image for the sanity account to ensure we have some branding, and I also wonder if this can be simplified a little - see inline comment.

Once approved, it would be great if you could open a PR against 2.x-next too: https://github.com/sanity-io/sanity/blob/2.x-next/packages/@sanity/base/src/datastores/user/createUserStore.ts#L49-L57

File: packages/sanity/src/core/store/_legacy/user/userStore.ts
Comment (binoy14) [reply packages/sanity/src/core/store/_legacy/user/userStore.ts:37]: oh ya that works and is much cleaner thanks!

**Review by rexxars — APPROVED** [review]
LGTM!

**Pull request merged**
